pkg/web/transactions.go

```diff
@@ -560,16 +577,35 @@ func (s *Server) RejectTransaction(c *gin.Context) {
 		return
 	}
 
-	c.Negotiate(http.StatusOK, gin.Negotiate{
-		Offered:  []string{binding.MIMEJSON, binding.MIMEHTML},
-		Data:     out,
-		HTMLName: "transaction_reject.html",
-	})
+	detailURL, _ := url.JoinPath("/transactions", transaction.ID.String(), "info")
+	// Set a cookie to show a toast message on the page redirect.
+	setToastCookie(c, "transaction_reject_success", "true", detailURL, s.conf.Auth.CookieDomain)
+
+	// If the content requested is HTML (e.g. the web-front end), then redirect the user
+	// to the transaction detail page.
+	if c.NegotiateFormat(binding.MIMEJSON, binding.MIMEHTML) == binding.MIMEHTML {
```

Comment:
Do we need to redirect users if a rejection request is successful? The reject modal appears when users click the `Reject` button the detail page, so they're all ready at the detail page. Once the modal is closed and if the request is successful, we should be able to display a toast message without needing to redirect them.

Users may also access the modal on the transaction list page. If we only want them to view the modal on the detail page, the UI may be updated to not display the reject modal on the transaction list page.

Reply:
I was following the same pattern as Accept, but certainly, we can do things differently for reject. The reason for the redirect is because we can't simply re-render the transaction detail page; but if you just need an AJAX 204 response back we can make that happen (or some other partial, is there a swap happening?)

Reply:
Ok, I'll work on this in SC-25929